interlok-aws-kinesis-sdk/src/main/java/com/adaptris/aws/kinesis/KinesisSDKStreamProducer.java

```diff
@@ -138,12 +174,71 @@ protected void doProduce(AdaptrisMessage msg, String endpoint) throws ProduceExc
     }
   }
 
-  private void doSend(AmazonKinesis kinesisClient, String endpoint, List <PutRecordsRequestEntry> putRecordsRequestEntryList){
-    PutRecordsRequest putRecordsRequest  = new PutRecordsRequest();
-    putRecordsRequest.setStreamName(endpoint);
-    putRecordsRequest.setRecords(putRecordsRequestEntryList);
-    PutRecordsResult putRecordsResult = kinesisClient.putRecords(putRecordsRequest);
-    log.trace("PutRecordResults: {}", putRecordsResult);
+  // According to the docs, we need to wait for the stream to be active
+  // https://docs.aws.amazon.com/streams/latest/dev/kinesis-using-sdk-java-create-stream.html
+  private void doAwaitStreamActive(AmazonKinesis kinesisClient, String endpoint) throws ProduceException {
```

Comment:
@telus-terchu Most of the line below are not covered by tests.
How much effort would it be to cover them, only if relevant?
Same for the same class in aws2.

Reply:
@sebastien-belin-adp I've added a test which covers the majority of the added code.

Reply:
Brillant, thanks